oracle.py

```diff
@@ -73,20 +75,22 @@ def insert_card(self, name, card):
     sql += ', '.join('?' for prop in Oracle.properties())
     sql += ')'
     values = [card.get(self.underscore2camel(property)) for property in Oracle.properties()]
-    self.database.execute(sql, values)
+    # self.database.execute commits after each statement, which we want to 
+    # avoid while inserting cards
+    self.database.database.execute(sql, values)
     id = self.database.value('SELECT last_insert_rowid()')
     for name in card.get('names', []):
-      self.database.execute('INSERT INTO card_name (card_id, name) VALUES (?, ?)', [id, name])
+      self.database.database.execute('INSERT INTO card_name (card_id, name) VALUES (?, ?)', [id, name])
```

Comment:
self.database.database could become self.database.connection, but I take your point about calling the function on Database's instance variable. I didn't want to change or rename the current execute+commit of my own accord, but adding a boolean param sounds like the best solution.